### PR TITLE
Remove some allocations

### PR DESCRIPTION
**Pull request by ljedrz**
This PR contains a few self-contained changes which are aimed at reducing the number of allocations.

### CODE REVIEW

File: circuit/program/src/data/record/to_bits.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ impl<A: Aleo> ToBits for Record<A, Plaintext<A>> {
         let data_bits_le = self
             .data
             .iter()
-            .flat_map(|(identifier, entry)| vec![identifier.to_bits_le(), entry.to_bits_le()])
+            .flat_map(|(identifier, entry)| [identifier.to_bits_le(), entry.to_bits_le()])
             .flatten()
```
Comment (zvolin) [inline circuit/program/src/data/record/to_bits.rs:29]: hard to say without benchmarks but wouldn't not introducing an array be faster?
```suggestion
            .flat_map(|(identifier, entry)| identifier.to_bits_le().chain(entry.to_bits_le()))
```

File: console/program/src/data/record/to_bits.rs [hunk 2]
```diff
@@ -23,7 +23,7 @@ impl<N: Network> ToBits for Record<N, Plaintext<N>> {
         let data_bits_le = self
             .data
             .iter()
-            .flat_map(|(identifier, entry)| vec![identifier.to_bits_le(), entry.to_bits_le()])
+            .flat_map(|(identifier, entry)| [identifier.to_bits_le(), entry.to_bits_le()])
```
Comment (zvolin) [inline console/program/src/data/record/to_bits.rs:26]: saa

File: circuit/program/src/data/record/to_bits.rs (hunk 1 above)
Comment (ljedrz) [reply circuit/program/src/data/record/to_bits.rs:29]: it's unlikely that there would be a difference in performance, but there is a more elegant way to do this; I'll add a new commit

**Comment by howardwu** [comment]
These are great changes. Note that #1225 will cause some breakages here. We also have some tech debt on `function ID` incoming from #1225. We need to abstract it into a method in `console` and `circuit`.

**Review by howardwu — APPROVED** [review]
LGTM, really nice.

**Pull request merged**